### PR TITLE
notification tests working

### PR DESCRIPTION
**Pull request by SarthakSingh31**

### CODE REVIEW

File: foodalert/test/test_notification copy.py [hunk 1]
```diff
@@ -0,0 +1,503 @@
+import os
```
Comment (abztrakt) [inline foodalert/test/test_notification copy.py]: test_notification copy, eh?

File: foodalert/test/test_notification.py [hunk 2]
```diff
@@ -328,126 +426,128 @@ def test_post_incomplete_payload(self):
...
 
-        with generate_twilio_mock() as mock:
-            response = self.client_4.post(
-                '/notification/',
-                data=proper_payload,
-                content_type='application/json'
-            )
-            self.assertEqual(response.status_code, 403)
+        response = client.post(
+            "/notification/0/",
```
Comment (abztrakt) [inline foodalert/test/test_notification.py]: That's an invalid id, right?

File: foodalert/test/test_notification.py [hunk 3]
```diff
@@ -328,126 +426,128 @@ def test_post_incomplete_payload(self):
...
-                data=valid_payload,
-                content_type='application/json'
-            )
-        self.assertEqual(response.status_code, 405)
+        client = create_client_with_mock_saml(
+            self.test_data[2]["host"],
+            [create_group, audit_group]
+        )
+        response = client.patch(
+            "/notification/0/",
```
Comment (abztrakt) [inline foodalert/test/test_notification.py]: Another 0!

File: foodalert/test/test_notification.py [hunk 4]
```diff
@@ -328,126 +426,128 @@ def test_post_incomplete_payload(self):
...
-        Attempts to patch a notification at id expect a 405
+        Attempts to patch a notification at id expect a 403
         """
-        response = self.client_1.delete(
+        client = create_client_with_mock_saml(
+            self.user1,
+            [create_group, audit_group]
+        )
+        response = client.delete(
                 "/notification/0/",
```
Comment (abztrakt) [inline foodalert/test/test_notification.py]: 0!

File: foodalert/test/test_notification.py [hunk 5]
```diff
@@ -461,8 +561,7 @@ def data_to_list_represent(self, data):
             }
 
     def data_to_detail_json(self, data):
```
Comment (abztrakt) [inline foodalert/test/test_notification.py]: Change the name so it reflects what's now being returned.

**Review by abztrakt — APPROVED** [review]
Fix the things we left comments on, and you're good to merge.

File: foodalert/test/test_notification copy.py (hunk 1 above)
Comment (SarthakSingh31) [reply foodalert/test/test_notification copy.py]: deleted

File: foodalert/test/test_notification.py (hunk 2 above)
Comment (SarthakSingh31) [reply foodalert/test/test_notification.py]: changed to 1

File: foodalert/test/test_notification.py (hunk 3 above)
Comment (SarthakSingh31) [reply foodalert/test/test_notification.py]: changed to 1

File: foodalert/test/test_notification.py (hunk 4 above)
Comment (SarthakSingh31) [reply foodalert/test/test_notification.py]: changed to 1

**Pull request merged**